src/cpu/aarch64/matmul/acl_matmul.hpp

```diff
@@ -92,8 +102,18 @@ struct acl_matmul_t : public primitive_t {
             VDISPATCH_MATMUL(!has_runtime_dims_or_strides(),
                     VERBOSE_RUNTIMEDIM_UNSUPPORTED);
 
-            CHECK(acl_matmul_utils::init_conf_matmul(
-                    amp_, src_md_, weights_md_, dst_md_, *desc(), *attr()));
+            if (weights_format_kind == format_kind::any) {
+                CHECK(acl_matmul_utils::init_conf_matmul_fixed_format(
+                        amp_, src_md_, weights_md_, dst_md_, *desc(), *attr()));
+            } else {
+#if DNNL_CPU_THREADING_RUNTIME == DNNL_RUNTIME_THREADPOOL
+                // to avoid seg. fault in case threadpool is enabled and its pointer is null
+                if (threadpool_utils::get_active_threadpool() == nullptr)
+                    return status::unimplemented;
+#endif
```

Comment:
Doesn't the change in `src/cpu/aarch64/acl_threadpool_scheduler.cpp` solve this problem?

Reply:
No, because the scheduler pointer is still null, the program can still crash.

Reply:
Could you remind, please, which line of code leads to a crash if this condition is removed?

Reply:
Ir crashes [here](https://github.com/oneapi-src/oneDNN/blob/e756ec6410baa453751a20a3932e5f6e415c2c3f/src/cpu/aarch64/acl_threadpool_scheduler.cpp#L116) when ACL calls _run_workloads_. This is related to this [question](https://github.com/oneapi-src/oneDNN/issues/1719).